samples/Kconfig.mcumgr_bt_ota_dfu

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+DT_CHOSEN_EXT_FLASH:= nordic,pm-ext-flash
+
+config NCS_SAMPLE_MCUMGR_BT_OTA_DFU
+	bool "MCUmgr OTA DFU over Bluetooth"
+	select MCUMGR
+	select MCUMGR_SMP_BT
+	imply MCUMGR_SMP_BT_CONN_PARAM_CONTROL
+	imply MCUMGR_CMD_IMG_MGMT
+	imply MCUMGR_CMD_OS_MGMT
+	imply MCUMGR_GRP_ZEPHYR_BASIC
+	imply MCUMGR_GRP_BASIC_CMD_STORAGE_ERASE
```

Comment:
Implying these options may result in build failure caused by undefined storage partition. The option is still set even if sample does not define storage partition, which leads to build failures.
Try `samples/bluetooth/peripheral_gatt_dm` built with: `west build -b nrf52840dk_nrf52840 -d new --pristine -- -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
` to replicate the problem.
I think we may need to ensure that the needed partition is defined when the option is enabled.

Reply:
fixed it using conditional enabling of the affected configurations